WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/PostsListPage.kt

```diff
@@ -20,17 +20,10 @@ object PostsListPage {
 
     // Workaround for cases when recyclerview id missing
     private val recyclerView: RecyclerView?
-        private get() {
+        get() {
             val pager = WPSupportUtils.getCurrentActivity().findViewById<ViewPager>(R.id.postPager)
             var recyclerView = pager.getChildAt(pager.currentItem)
                 .findViewById<View>(R.id.recycler_view) as RecyclerView
-            if (recyclerView == null) {
-                // Workaround for cases when recyclerview id missing
-                recyclerView = ((pager.getChildAt(pager.currentItem)
-                    .findViewById<View>(R.id.ptr_layout) as ViewGroup).getChildAt(0) as ViewGroup).getChildAt(
-                    0
-                ) as RecyclerView
-            }
```

Comment:
I'm sorry for approving and then dismissing the review @jostnes, there is something I noticed about this part already after I approved the PR.

In `wPScreenshotTest` I noticed that `editBlogPost` is disabled (making the removed part of code not really affect anything), but then I noticed a [comment](https://github.com/wordpress-mobile/WordPress-Android/blob/update-e2e-files-to-kotlin/WordPress/src/androidTest/java/org/wordpress/android/ui/screenshots/WPScreenshotTest.java#L78) above `editBlogPost` saying that it's disabled because it fails on CI.

This made me think that it's still executed locally (this is where screenshot tests are usually executed, IINM). After trying to execute it, it appeared that the `recyclerView` locator that was left after removal from above did not work, and additionally it failed the run, because of "can't cast `null` to RecyclerView". Returning back the removed code and applying a minor change (moving `as RecyclerView` to `return`) did the trick - the secondary locator now is called:

```kotlin
    // Workaround for cases when recyclerview id missing
    private val recyclerView: RecyclerView?
        get() {
            val pager = WPSupportUtils.getCurrentActivity().findViewById<ViewPager>(R.id.postPager)
            var recyclerView = pager.getChildAt(pager.currentItem)
                .findViewById<View>(R.id.recycler_view)
            if (recyclerView == null) {
                // Workaround for cases when recyclerview id missing
                recyclerView = ((pager.getChildAt(pager.currentItem)
                    .findViewById<View>(R.id.ptr_layout) as ViewGroup).getChildAt(0) as ViewGroup).getChildAt(
                    0
                )
            }
            return recyclerView as RecyclerView
        }
```

Reply:
👋 @pachlava @jostnes !

Actually, I think the issue was that the conversion of this file from Java to Kotlin, which left the `RecyclerView` cast as non-null, instead of it being nullable (I mean here `findViewById<View>(R.id.recycler_view)`).

As such, just adding a `?` on `RecyclerView` and making it `RecyclerView?` would all that needed happened here (I think). 

End result:

```
    // Workaround for cases when recyclerview id missing
    private val recyclerView: RecyclerView?
        get() {
            val pager = WPSupportUtils.getCurrentActivity().findViewById<ViewPager>(R.id.postPager)
            var recyclerView = pager.getChildAt(pager.currentItem)
                .findViewById<View>(R.id.recycler_view) as RecyclerView?
            if (recyclerView == null) {
                // Workaround for cases when recyclerview id missing
                recyclerView = ((pager.getChildAt(pager.currentItem)
                    .findViewById<View>(R.id.ptr_layout) as ViewGroup).getChildAt(0) as ViewGroup).getChildAt(
                    0
                ) as RecyclerView
            }
            return recyclerView
        }
```

Let me know if that makes sense to you. 🙏

Reply:
@pachlava @ParaskP7 Thanks for pointing this out! I removed it as suggested by Android Studio as the condition was never going to be met. After the change, I saw that everything was still passing, so I incorrectly assumed that everything was good. 

I've reverted the if condition and made RecyclerView nullable by adding the `?` as suggested. Updated in https://github.com/wordpress-mobile/WordPress-Android/pull/18071/commits/c2c88c71aa63519c0fe5c2d3488aad62cb92068f

Reply:
Awesome, thanks for the update @jostnes ! 🙇 ❤️ 🚀